esmvaltool/preprocessor/_area_pp.py

```diff
@@ -0,0 +1,110 @@
+"""
+Area operations on data cubes
+
+Allows for selecting data subsets using certain latitude and longitude bounds;
+selecting geographical regions; constructing area averages; etc.
+"""
+import iris
+
+
+# slice cube over a restricted area (box)
+def area_slice(mycube, long1, long2, lat1, lat2):
+    """
+    Subset a cube on area
+
+    Function that subsets a cube on a box (long1,long2,lat1,lat2)
+    This function is a restriction of masked_cube_lonlat();
+    Returns a cube
+    """
+    # Converts Negative longitudes to 0 -> 360. standard
+    if long1 < 0.:
+        long1 += 360.
+    if long2 < 0.:
+        long2 += 360.
+
+    if long2 < long1:
+        # if you want to look at a region both sides of
+        # the zero longitude ie, such as the Atlantic Ocean!
+        sublon = iris.Constraint(
+            longitude=lambda cell: not ((float(long1) >= cell)
+                                        * (cell >= float(long2))))
+    else:
+        sublon = iris.Constraint(
+            longitude=lambda cell: float(long1) <= cell <= float(long2))
+    sublat = iris.Constraint(
+        latitude=lambda cell: float(lat1) <= cell <= float(lat2))
+    region_subset = mycube.extract(sublon & sublat)
+    return region_subset
+
+
+# get zonal means
+def zonal_means(mycube, coord1, mean_type):
```

Comment:
Use names that are easy to read, e.g.:
`def zonal_means(cube, coordinate, mean_type):`